lib/components/logic/defaultMatcher.js

```diff
@@ -0,0 +1,5 @@
+import deepEquals from "../../utils/deepEquals"
+
+const defaultMatcher = (expression, value) => deepEquals(value, expression, { partial: true });
```

Comment:
Can the default still be strict equality and we allow the user to pass in a list of supported matchers?